yaqd_andor/_andorsdk2_ixon.py

```diff
@@ -0,0 +1,277 @@
+                raise ValueError(str(self.errorlookup(code)))
+            code = self.sdk.SetTemperature(self.sensor_temp_control)
+            if code != 20002:
+                raise ValueError(str(self.errorlookup(code)))
+            self.logger.info(f"Sensor is cooling.  Target temp is {self.sensor_temp_control} C.")
+            self._loop.run_in_executor(None, self._check_temp_stabilized)
+        else:
+            code, self.sensor_temp = self.sdk.GetTemperature()
+            self.logger.info(f"Sensor is not cooled.  Current temp is {self.sensor_temp} C.")
+
+    def _check_temp_stabilized(self):
+        code, self.sensor_temp = self.sdk.GetTemperature()
+        diff = float(self.sensor_temp_control) - float(self.sensor_temp)
+        while np.abs(diff) > 1.0:  # this is a tolerance, and is subject to change
+            self.logger.info(
+                f"Sensor is cooling.  Target: {self.sensor_temp_control} C.  Current: {self.sensor_temp:0.2f} C."
+            )
+            sleep(3)
+            code, self.sensor_temp = self.sdk.GetTemperature()
+            diff = float(self.sensor_temp) - float(self.sensor_temp_control)
+
+        self.logger.info("Sensor temp is stabilized.")
+
+    def set_spec_position(self, spec_position):
+        # Sets the CCD mapping to a new spectrometer position (nm)
+        if self.has_mono:
+            self.spec_position = float(spec_position)
+            self._gen_mappings()
+            self.logger.info(f"new spectrometer position: {self.spec_position} nm")
+        else:
+            self.logger.info(
+                f"Attempt to change spectrometer position when monochromator portion not enabled."
+            )
+
+    def get_spec_position(self):
+        # Gets the CCD mapping to the spectrometer position (nm)
+        if self.has_mono:
+            return self.spec_position
+        else:
+            self.logger.info(
+                f"Attempt to change spectrometer position when monochromator portion not enabled."
+            )
+            return float(0.00)
+
+    def set_exposure_time(self, exposure_time):
+        # Sets the exposure time in seconds (float)
+        self.stop_update = True
+        while self._busy == True:
+            sleep(0.10)
+        code = self.sdk.SetExposureTime(float(exposure_time))
+        if code != 20002:
+            raise ValueError(str(self.errorlookup(code)))
+        else:
+            self.exposure_time = exposure_time
+            self.logger.info(f"New exposure time is {self.exposure_time} sec.")
+        self.stop_update = False
+        return code
+
+    def get_exposure_time(self):
+        # Gets the exposure time in seconds (float)
+        return self.exposure_time
+
+    def close(self):
+        # stop loop
+        self.stop_update = True
+        while self._busy == True:
+            sleep(0.10)
+        self.sdk.SetShutter(int(0), int(2), int(100), int(100))
+        sleep(0.50)
+        self.sdk.CoolerOFF()
```

Comment:
I do not believe the cooler turns off upon close.

How often would a camera need to restart?   In principle we should not have to do that often.   

Is there a risk of damage to the camera if the TE cooling was suddenly removed at -40C due to power off?

Reply:
> How often would a camera need to restart? In principle we should not have to do that often.
True for experiments.  But for testing restarts are frequent.  Regardless, I am failing to see the point of watching the temp warm up before closing.

> Is there a risk of damage to the camera if the TE cooling was suddenly removed at -40C due to power off?

I see no risk of damage.  In any case, your approach literally tells the cooler to turn off; there is no "slow warming" mechanism in place to gradually return to RT.

Reply:
slow warming is the next 2 lines of code.   I was concerned of thermal stresses but i have not seen any documentation claiming a gradual warmup must be done to protect the ccd.

Reply:
The two lines of code you mention set the temperature to 25 C and wait for it to get there. Not at all clear to me that those lines of code do anything that wouldn't be done by just powering down the cooler/camera. Since there is no evidence/documentation that this procedure is necessary, and since it adds complexity/time to the shutdown procedure, is it reasonable to leave it out for now?  

Note that if you feel you want to play with temperature ramping, it will be possible to do this client-side for testing.